src/libsyntax_ext/concat.rs

```diff
@@ -51,17 +52,15 @@ pub fn expand_syntax_ext(
                 }
             },
             _ => {
-                let mut err = cx.struct_span_err(e.span, "expected a literal");
-                let snippet = cx.codemap().span_to_snippet(e.span).unwrap();
-                err.span_suggestion(
-                    e.span,
-                    "you might be missing a string literal to format with",
-                    format!("\"{{}}\", {}", snippet),
-                );
-                err.emit();
+                missing_literal.push(e.span);
             }
         }
     }
+    if missing_literal.len() > 0 {
+        let mut err = cx.struct_span_err(missing_literal, "expected a literal");
+        err.note("only `&str` literals can be passed to `concat!()`");
```

Comment:
Most literals can be passed to `concat!`, not just `&str`.

Reply:
True, but the wording "only literals..." can be harder to understand to newcomers... What do you think of
```
only literals (like `"foo"`, `42` and `3.14`) can be passed to `concat!()`
```